src/protocol.rs

```diff
@@ -864,4 +892,13 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn padding() {
+        let query = ObliviousDoHMessagePlaintext::new(&[], 0);
+        assert_eq!(query.padding_len(), 0);
+
+        let query = ObliviousDoHMessagePlaintext::new(&[], 2);
+        assert_eq!(query.padding_len(), 2);
```

Comment:
Could we add tests that serialize a message and check that there are the appropriate amount of padding bytes, all zero, too?

Reply:
Add a test to cover the non-zero padding check logic.